src/dotty/tools/dotc/core/pickling/PickleFormat.scala

```diff
@@ -0,0 +1,453 @@
+
+Macro-format:
+
+  File          = Header majorVersion_Nat minorVersion_Nat nameTable_Length Name* Section*
+  Header        = "5CA1AB1F"
+
+  Section       = NameRef Length Bytes
+  Length        = Nat                    // length of rest of entry in bytes
+
+  Name          = UTF8	        Length UTF8-CodePoint*
+                  QUALIFIED     Length qualified_NameRef selector_NameRef
+                  SIGNED        Length original_NameRef resultSig_NameRef paramSig_NameRef*
+                  EXPANDED      Length original_NameRef
+                  MODULECLASS   Length module_NameRef
+                  SUPERACCESSOR Length accessed_NameRef
+                  DEFAULTGETTER Length method_NameRef paramNumber_Nat
+                  ...
+
+  NameRef       = Nat                    // ordinal number of name in name table, starting from 1.
+
+Note: Unqualified names in the name table are strings. The context decides whether a name is
+a type-name or a term-name. The same string can represent both.
+
+Standard-Section: "ASTs" Tree*
+
+  Tree          = PACKAGE       Length Path Tree*
+                  Stat
+
+  Stat          = Term
+                  VALDEF        Length NameRef Type rhs_Tree Modifier*
+                  DEFDEF        Length NameRef TypeParam* Params* return_Type rhs_Tree
+                                       Modifier*
+                  TYPEDEF       Length NameRef (Type | Template) Modifier*
+                  IMPORT        Length qual_Term Selector*
+
+  TypeParam     = TYPEPARAM     Length NameRef Type Modifier*
+  Params        = PARAMS        Length Param*
+  Param         = PARAM         Length NameRef Type Modifier
+  Selector      = IMPORTED      Length name_NameRef
+                  RENAMED       Length from_NameRef to_NameRef
+
+  Term          = Path
+                  IDENT                Type NameRef
+                  SELECT               qual_Term possiblySigned_NameRef
+                  SUPER         Length this_Term mixinTrait_Type?
+                  APPLY         Length fn_Term arg_Term*
+                  TYPEAPPLY     Length fn_Term arg_Term*
+                  NEW           Length cls_Type
+                  PAIR          Length left_Term right_Term
+                  TYPED         Length expr_Term ascription_Type
+                  NAMEDARG      Length paramName_NameRef arg_Term
+                  ASSIGN        Length lhs_Term rhs_Term
+                  BLOCK         Length expr_Term Stat*
+                  IF            Length cond_Term then_Term else_Term
+                  CLOSURE       Length meth_Term target_Type env_Term*
+                  MATCH         Length sel_Term CaseDef*
+                  RETURN        Length meth_ASTRef expr_Term?
+                  TRY           Length expr_Term CaseDef* finalizer_Term?
+                  THROW         Length expr_Term
+                  SEQLITERAL    Length elem_Term*
+                  JSEQLITERAL   Length elem_Term*
```

Comment:
If JSEQLITERAL is a Java version of the SEQLITERAL I think it should not be in the format. This would tie the format to a particular back-end which, I believe, we want to avoid. 

What would the scala.js backend or an LLVM backend do with this?